server/src/promo-dialog.js

```diff
@@ -0,0 +1,51 @@
+const React = require("react");
+const PropTypes = require("prop-types");
+const { Localized } = require("fluent-react/compat");
+const sendEvent = require("./browser-send-event.js");
+
+exports.PromoDialog = class PromoDialog extends React.Component {
+  constructor(props) {
+    super(props);
+  }
+
+  render() {
+    if (this.props.display) {
+      this.props.promoOpen(this.props.promoType);
```

Comment:
Given how `promoOpen` is implemented, I think this should only happen on `componentDidMount` – at render time we don't know what's in the document.

Reply:
componentDidMount have this.props.display as undefined on first render, moved the check inside componentDidUpdate